src/full/Agda/Syntax/Abstract/Views.hs

```diff
@@ -1,4 +1,6 @@
 {-# LANGUAGE CPP                       #-}
+{-# LANGUAGE GADTs                     #-}
+{-# LANGUAGE NoMonoLocalBinds          #-}
```

Comment:
So you might be wondering why enabling `NoMonoLocalBinds` is necessary here. The confusing reality is that enabling `GADTs` implies also enabling `MonoLocalBinds`, which causes dozens of functions below to have monomorphic inferred type signatures (even in the presence of `NoMonomorphismRestriction`!) Thus we need to have `NoMonoLocalBinds` appear after `GADTs` to counter this behavior.